### PR TITLE
Add new my markets filter, reconfigure existing filters for greater maintainability

### PR DESCRIPTION
**Pull request by KadenZipfel**
Closes: https://github.com/protofire/omen-exchange/issues/2017

- Added a 'Ended recently' default filter to my markets
- Reconfigured existing market filters for simpler maintainability
- Still plan to add more advanced my markets filters, but will require some deeper work on the subgraph so this should be a useful quick fix

Testing: 
- [ ] All filters in all states work as expected, all the time
  - Trying going between every different filter and checking that the ordering is correct

### CODE REVIEW

**Review by hexyls — COMMENTED** [review]
When trying to filter "My Markets" + "Ended recently" + "Scalar markets" I have 1 categorical market in the list, can you reproduce?

**Review by pimato — CHANGES_REQUESTED** [review]
Also we should probably keep track of the previously picked/default sort type when going back from market details view to market overview:

https://user-images.githubusercontent.com/5337809/127847876-0c9544cf-d17f-46c0-b3d1-c5fcabeae90d.mov

> When trying to filter "My Markets" + "Ended recently" + "Scalar markets" I have 1 categorical market in the list, can you reproduce?

can reproduce:

https://user-images.githubusercontent.com/5337809/127848324-b54a1750-d364-48e6-a309-f8a478af17e7.mov

**Comment by KadenZipfel** [comment]
> When trying to filter "My Markets" + "Ended recently" + "Scalar markets" I have 1 categorical market in the list, can you reproduce?

Yeah, sadly we can't query by market type yet for my markets, but I've added an subgraph issue for how to make this possible: https://github.com/protofire/omen-subgraph/issues/184

**Comment by KadenZipfel** [comment]
> Also we should probably keep track of the previously picked/default sort type when going back from market details view to market overview:

I've set it to keep the default sort type when we go back now. It can also be set to go to the previously picked filter, but would require a little extra work... Should I make an issue for that?

**Review by hexyls — APPROVED** [review]
LGTM 🙌

**Comment by pimato** [comment]
> > Also we should probably keep track of the previously picked/default sort type when going back from market details view to market overview:
> 
> I've set it to keep the default sort type when we go back now. It can also be set to go to the previously picked filter, but would require a little extra work... Should I make an issue for that?

I´ve created an issue for it: https://github.com/protofire/omen-exchange/issues/2150

File: app/src/pages/market_sections/market_home_container.tsx [hunk 1]
```diff
@@ -140,26 +147,30 @@ const MarketHomeContainer: React.FC = () => {
...
+      sortIndex = 2
+      sortParam = 'usdLiquidityParameter'
+    } else if (sortRoute === 'newest') {
+      sortIndex = 3
+      sortParam = 'creationTimestamp'
+    } else if (sortRoute === 'ending') {
+      sortIndex = 4
+      sortParam = 'openingTimestamp'
+      sortDirection = 'asc'
+    }
```
Comment (Mi-Lan) [inline app/src/pages/market_sections/market_home_container.tsx:173]: @KadenZipfel Wondering is it maybe more efficient to make this into switch/case statement?

**Review by Mi-Lan — APPROVED** [review]
@KadenZipfel Good job man! Tested it out works as it should! Nice addition with the new ended recently filter. Made just one small comment.

**Review by pimato — APPROVED** [review]

File: app/src/pages/market_sections/market_home_container.tsx [hunk 2]
```diff
@@ -304,16 +341,18 @@ const MarketHomeContainer: React.FC = () => {
...
+          route += '/24h-volume'
+        } else if (filter.sortBy === 'usdVolume') {
+          route += '/volume'
+        } else if (filter.sortBy === 'creationTimestamp') {
+          route += '/newest'
+        } else if (filter.sortBy === 'openingTimestamp') {
+          route += '/ending'
+        } else if (filter.sortBy === 'usdLiquidityParameter') {
+          route += '/liquidity'
+        }
```
Comment (Mi-Lan) [inline app/src/pages/market_sections/market_home_container.tsx:355]: @KadenZipfel You are gonna kill me 🗡️  Should we also change it here...Should have outlined all at the start...

**Review by Frankaus — APPROVED** [review]
All the filters work smoothly, great job and great addition on the filter under My Markets 👍

File: app/src/pages/market_sections/market_home_container.tsx (hunk 2 above)
Comment (KadenZipfel) [reply app/src/pages/market_sections/market_home_container.tsx:355]: Good catch! Will fix

**Review by Mi-Lan — APPROVED** [review]
@KadenZipfel GJ!

**Pull request merged**